clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -888,11 +889,18 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 // treat subscribing to empty topic list as the same as unsubscribing
                 this.unsubscribe();
             } else {
+                IllegalArgumentException iae = new IllegalArgumentException("Detected issues with topic names in topic partitions to assign:");
                 for (String topic : topics) {
-                    if (topic == null || topic.trim().isEmpty())
-                        throw new IllegalArgumentException("Topic collection to subscribe to cannot contain null or empty topic");
+                    try {
+                        Topic.validate(topic);
+                    } catch (Exception e) {
+                        iae.addSuppressed(e);
+                    }
```

Comment:
instead of iterating all the topics, maybe we can fail after the first topic failure itself?

Reply:
I think that reporting all errors at once provides a better user experience. Normally the list is not that long and validating all topics do not cause much overhead. Is there another reason behind your suggestion?

Reply:
Agree. I was thinking of extracting code to throwIfTopicNotValid() method. so that it can be used at both the places.

Reply:
Just throwing on the first is probably fine. Alternatively, if you want to list them all, I'd suggest just through them and collecting them into a collection rather than using suppressed exceptions.